clients/src/main/java/org/apache/kafka/clients/admin/ElectPreferredLeadersResult.java

```diff
@@ -34,9 +34,12 @@
  * The result of {@link AdminClient#electPreferredLeaders(Collection, ElectPreferredLeadersOptions)}
  *
  * The API of this class is evolving, see {@link AdminClient} for details.
+ *
+ * @deprecated Since TDB. Use {@link AdminClient#electLeaders(ElectionType, Collection, ElectLeadersOption)}.
  */
 @InterfaceStability.Evolving
-public class ElectPreferredLeadersResult {
+@Deprecated
+final public class ElectPreferredLeadersResult {
```

Comment:
I'll remove the `final`.

Regarding code duplication. We have 3 options:
1. `ElectLeadersResult extends ElectPreferredLeadersResult`. This means that a supported type is a deprecated type. Eventually, we want to remove `ElectPreferredLeadersResult` which means the type hierarchy for `ElectLeadersResult` will change in a backwards incompatible way.
2. `ElectPreferredLeadersResult extends ElectLeadersResult`. This is a change in the type hierarchy for `ElectPreferredLeadersResult` but maybe this is okay. This also means that `ElectLeadersResult` is not final which I am trying to avoid.
3. Share some code through composition. I am not 100% sure how this would work but let me investigate. Maybe `ElectPreferredLeadersResult` can be a wrapper for `ElectLeadersResult` that delegates to it.

Reply:
`electPreferredLeaders` now uses `electLeaders`. I implemented `ElectPreferredLeadersResult` using option 3.